### PR TITLE
Optimize build steps in Dockerfile to reduce image size

### PR DESCRIPTION
**Pull request by bitfag**
Install dependencies, compile, install, cleanup dependencies now made in
one step. This allows to significantly reduce docker image size from
1.45GB to 832MB.

Closes: #569

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -27,13 +29,8 @@ RUN \
         python3-dev \
         python3-pip \
     && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* && \
```
Comment (zxcat) [inline Dockerfile]: > \-    apt-get clean && \
\-    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* && \

Is it ok to remove this lines completely?

Comment (bitfag) [reply Dockerfile]: You can find the same lines in the cleanup section around `apt-get autoremove`

**Comment by bitfag** [comment]
I can do a separate Dockerfile for sure. The question is, how it should be by default? Is default Dockerfile intended for regular cached rebuilds or for a production use?

**Comment by afalaleev** [comment]
Existing cached layers have several advantages:
- They are used on the rebuild of a docker-image
- They are used on the pushing of a docker-image to the docker-hub
- They are used on the upgrading of a docker-image on a target node.

It is optimized to update the golosd.

For example, if you are using the public version of the doker-image, upgrading to a new version will require to download a layer with a rebuilded version of a golosd without necessity to download the full docker-image.

If we will make your version of Dockerfile by default, it will be very strange that the public version of goloschain/golos:XXX will be assembled not from root-dockerfile.

So, I can accept your pull-request, but if it will be an additional dockerfile, not main. Because it is a good version for manually building of a docker-image.

And it will be really nice if you will create a dockerfile for the LOW_MEMORY_NODE.

Also, I can accept an additional description to the build instruction, if you want do it. In any case we will add information about your Dockerfile to the build instruction.

**Comment by bitfag** [comment]
OK, I'll prepare a separate Dockerfile.

About Dockerfile for the LOW_MEMORY_NODE: on what Dockerfile it should be based? `-small`?

**Pull request merged**